include/sys/zstd/zstd.h

```diff
@@ -110,29 +110,27 @@ void zfs_zstd_cache_reap_now(void);
  *
  *      32      24      16      8       0
  *      +-------+-------+-------+-------+
- *      | level |   0   |    version    |
+ *      | level |  v0   |  v1   |  v2   |
  *      +-------+-------+-------+-------+
  *
  * ...and then, after being run through BE_32(), serializing this out to
- * disk (BS16 indicates an order swap between the two bytes):
+ * disk:
  *
  *      32      24      16      8       0
  *      +-------+-------+-------+-------+
- *      | BS16(version) |   0   | level |
+ *      |  v2   |  v1   |  v0   | level |
  *      +-------+-------+-------+-------+
  *
  * while on big-endian systems, since BE_32() is a noop there, both in
  * memory and on disk, we wind up with:
  *
  *      32      24      16      8       0
  *      +-------+-------+-------+-------+
- *      | level | BS16(version) |   0   |
+ *      | level |  v2   |  v1   |  v0   |
  *      +-------+-------+-------+-------+
  *
- * (The "0" byte is part of the version field, but will remain 0 until
- * version 6.55.36, so it is useful both for illustration purposes
- * and for the "get" code below to know which possible layout of the
- * header we're dealing with.)
+ * (v0 is always 0 until version exceeds 6.55.35. v1 and v2 are the other
+ * two bytes of the "version" data.)
```

Comment:
I think that this notation is internally consistent.  The reasons I find it so confusing are:
1. "v0" refers to the high byte and "v2" refers to the low byte, but 0 < 2.  Presuming that you would find it confusing for "v0" to refer to the low byte, perhaps Vhi, Vmid, Vlow would be more clear?
2. The diagram uses the same notation as the ones in spa.h, which shows individual bits of a logical word, i.e. the register contents, which is the same regardless of endianness.  However, this diagram is not showing the bits of a word, it's showing bytes in memory/ondisk.  It's kind of confusing that the last byte appears first in the diagram.  But since you're manipulating it as a 32-bit word (not individual bytes), maybe this is helpful?

Reply:
1. Yeah, I'd prefer to not use v0 for the low byte, for similar reasons to your not liking v0 for the highest byte. Vhi et al WFM.
2. I have absolutely no preference on the ordering, the current choice is indeed just an artifact of plundering the ASCII art from spa.h and chopping 32 bits off. If it seems clearer to go 0...32 instead of 32...0, done.

I've also added example data to the comment, because I was having trouble following whether it was correct, and if the author can't follow the diagram, it needs more clarity.